### PR TITLE
Add additional queries to 'Other Compliance Checks'

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist

1. [x] eks_cluster_endpoint_public_access_restircted
2. [x] guardduty_centrally_configured
3. [x] iam_role_cross_account_readonlyaccess_policy
4. [x] iam_securityaudit_role
5. [x] rds_db_instance_connections_encryption_enabled
6. [x] vpc_in_more_than_one_region
7. [x] vpc_subnet_multi_az_enabled
8. [x] workspaces_workspace_volume_encryption_enabled
9. [x] autoscaling_ec2_launch_configuration_no_sensitive_data
10. [x] lambda_function_restrict_public_url
11. [x] lambda_function_variables_no_sensitive_data
12. [x] iam_policy_custom_no_permissive_role_assumption
13. [x] vpc_subnet_public_and_private

### CODE REVIEW

File: conformance_pack/iam.sp [hunk 1]
```diff
@@ -1919,3 +1939,84 @@ query "iam_role_unused_60" {
...
+    )
+    select
+      r.arn as resource,
+      case
+        when ar.arn is null then 'skip'
+        when c.arn is not null then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when ar.arn is null then r.title || ' does have ReadOnlyAccess policy attached.'
```
Comment (rajlearner17) [inline conformance_pack/iam.sp:1975]:
```suggestion
        when ar.arn is null then r.title || ' not associated with ReadOnlyAccess policy.'
```

File: conformance_pack/iam.sp [hunk 2]
```diff
@@ -1919,3 +1939,84 @@ query "iam_role_unused_60" {
...
+        when c.arn is not null then r.title || ' have ReadOnlyAccess with cross account access.'
```
Comment (rajlearner17) [inline conformance_pack/iam.sp:1976]:
```suggestion
        when c.arn is not null then r.title || ' associated with ReadOnlyAccess cross account access.'
```

File: conformance_pack/iam.sp [hunk 3]
```diff
@@ -1919,3 +1939,84 @@ query "iam_role_unused_60" {
...
+        else r.title || ' does not have ReadOnlyAccess with cross account access.'
```
Comment (rajlearner17) [inline conformance_pack/iam.sp:1977]:
```suggestion
        else r.title || ' associated ReadOnlyAccess without cross account access.'
```

File: conformance_pack/guardduty.sp [hunk 4]
```diff
@@ -60,6 +60,16 @@ control "guardduty_no_high_severity_findings" {
   })
 }
 
+control "guardduty_centrally_configured" {
+  title       = "GuardDuty Detector should should be centrally configured"
```
Comment (rajlearner17) [inline conformance_pack/guardduty.sp:64]:
```suggestion
  title       = "GuardDuty Detector should be centrally configured"
```

File: conformance_pack/vpc.sp [hunk 5]
```diff
@@ -1578,4 +1608,100 @@ query "vpc_security_group_allows_ingress_authorized_ports" {
...
+    )
+    select
+      arn as resource,
+      case
+        when (select count(*) from vpc_region_list) > 1  then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when (select count(*) from vpc_region_list) > 1  then 'VPCs exist in ' || (select count(*) from vpc_region_list) || ' regions.'
+        else 'VPC  does not exist in more than one region.'
```
Comment (rajlearner17) [inline conformance_pack/vpc.sp:1632]: @khushboo9024 if there are NO subnets exists, then we can reason 
```
  ALARM: test_peering no subnet exists. ..................................................................................... us-east-2 123456781234
```

File: conformance_pack/vpc.sp [hunk 6]
```diff
@@ -373,6 +373,36 @@ control "vpc_security_group_allows_ingress_to_oracle_ports" {
...
+  title       = "VPCs should exist in multiple regions"
+  description = "This control checks whether there are VPCs present in multiple regions."
+  query       = query.vpc_in_more_than_one_region
+
+  tags = merge(local.conformance_pack_vpc_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "vpc_subnet_multi_az_enabled" {
```
Comment (rajlearner17) [inline conformance_pack/vpc.sp:386]: @khushboo9024 this should compliment to `vpc_in_more_than_one_region`, we may think to remove `vpc_in_more_than_one_region`

File: conformance_pack/lambda.sp [hunk 7]
```diff
@@ -360,6 +380,60 @@ query "lambda_function_use_latest_runtime" {
...
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when url_config is null then 'info'
+        when url_config ->> 'AuthType' = 'AWS_IAM' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when url_config is null then title || ' does not has a URL config.'
```
Comment (rajlearner17) [inline conformance_pack/lambda.sp:393]:
```suggestion
        when url_config is null then title || ' having no URL config.'
```
Does this make more sense?

File: conformance_pack/lambda.sp [hunk 8]
```diff
@@ -360,6 +380,60 @@ query "lambda_function_use_latest_runtime" {
...
+        when url_config ->> 'AuthType' = 'AWS_IAM' then title || ' restric public function URL.'
```
Comment (rajlearner17) [inline conformance_pack/lambda.sp:394]:
```suggestion
        when url_config ->> 'AuthType' = 'AWS_IAM' then title || ' restricts public function URL.'
```

File: conformance_pack/lambda.sp [hunk 9]
```diff
@@ -360,6 +380,60 @@ query "lambda_function_use_latest_runtime" {
...
+        else title || ' have public function URL.'
```
Comment (rajlearner17) [inline conformance_pack/lambda.sp:395]:
```suggestion
        else title || ' public function URL configured.'
```

**Review by rajlearner17 — APPROVED** [review]
LGTM

File: conformance_pack/iam.sp [hunk 10]
```diff
@@ -572,6 +572,36 @@ control "iam_policy_no_full_access_to_kms" {
   })
 }
 
+control "iam_role_cross_account_readonlyaccess_policy" {
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:575]:
```suggestion
control "iam_role_cross_account_read_only_access_policy" {
```

File: conformance_pack/iam.sp [hunk 11]
```diff
@@ -572,6 +572,36 @@ control "iam_policy_no_full_access_to_kms" {
...
+  title       = "IAM roles should not have read only access for external AWS accounts"
+  description = "Ensure IAM Roles do not have ReadOnlyAccess access for external AWS account. The AWS-managed ReadOnlyAccess policy carries a high risk of potential data leakage, posing a significant threat to customer security and privacy."
+  query       = query.iam_role_cross_account_readonlyaccess_policy
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:578]:
```suggestion
  query       = query. iam_role_cross_account_read_only_access_policy
```

File: conformance_pack/iam.sp [hunk 12]
```diff
@@ -1919,3 +1949,125 @@ query "iam_role_unused_60" {
       aws_iam_role;
   EOQ
 }
+
+query "iam_role_cross_account_readonlyaccess_policy" {
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:1953]:
```suggestion
query "iam_role_cross_account_read_only_access_policy" {
```

File: conformance_pack/iam.sp [hunk 13]
```diff
@@ -1919,3 +1949,125 @@ query "iam_role_unused_60" {
...
+      end as reason
+      ${replace(local.common_dimensions_qualifier_global_sql, "__QUALIFIER__", "r.")}
+    from
+      aws_iam_role as r
+      left join read_only_access_roles as ar on r.arn = ar.arn
+      left join read_only_access_roles_with_cross_account_access as c on c.arn = r.arn;
+  EOQ
+}
+
+query "iam_securityaudit_role" {
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:1997]:
```suggestion
query "iam_security_audit_role" {
```

File: conformance_pack/iam.sp [hunk 14]
```diff
@@ -1919,3 +1949,125 @@ query "iam_role_unused_60" {
...
+  sql = <<-EOQ
+    with securityaudit_role_count as(
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:1999]:
```suggestion
    with security_audit_role_count as(
```

File: conformance_pack/iam.sp [hunk 15]
```diff
@@ -1919,3 +1949,125 @@ query "iam_role_unused_60" {
...
+        else 'alarm'
+      end as status,
+      case
+        when count = 1 then 'SecurityAudit policy attached to 1 role.'
+        when count > 1 then 'SecurityAudit policy attached to ' || count || ' roles.'
+        else 'SecurityAudit policy not attached to any role.'
+      end as reason
+      ${local.common_dimensions_global_sql}
+    from
+      securityaudit_role_count;
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:2029]:
```suggestion
      security_audit_role_count;
```

File: conformance_pack/iam.sp [hunk 16]
```diff
@@ -572,6 +572,36 @@ control "iam_policy_no_full_access_to_kms" {
...
+  title       = "IAM roles should not have read only access for external AWS accounts"
+  description = "Ensure IAM Roles do not have ReadOnlyAccess access for external AWS account. The AWS-managed ReadOnlyAccess policy carries a high risk of potential data leakage, posing a significant threat to customer security and privacy."
+  query       = query.iam_role_cross_account_readonlyaccess_policy
+
+  tags = merge(local.conformance_pack_iam_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "iam_securityaudit_role" {
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:585]:
```suggestion
control "iam_security_audit_role" {
```

File: conformance_pack/iam.sp [hunk 17]
```diff
@@ -572,6 +572,36 @@ control "iam_policy_no_full_access_to_kms" {
...
+  title       = "IAM Security Audit role should be created to conduct security audits"
+  description = "Ensure IAM Security Audit role is created. By creating an IAM role with a security audit policy, a distinct segregation of responsibilities is established between the security team and other teams within the organization."
+  query       = query.iam_securityaudit_role
```
Comment (madhushreeray30) [inline conformance_pack/iam.sp:588]:
```suggestion
  query       = query. iam_security_audit_role
```

**Review by madhushreeray30 — COMMENTED** [review]
@khushboo9024 can you please have a look at the comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]
LGTM 👍

**Review by misraved — APPROVED** [review]

**Pull request merged**